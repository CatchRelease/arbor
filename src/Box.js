import styled from '@emotion/styled';
import {
  boxShadow,
  color,
  display,
  space,
  width,
  height,
  maxWidth,
  textAlign
} from 'styled-system';

const Box = styled.div`
  box-sizing: border-box;
  ${boxShadow};
  ${color};
  ${display};
  ${height};
  ${maxWidth};
  ${space};
  ${textAlign};
  ${width};
`;

Box.propTypes = {
  ...boxShadow.propTypes,
  ...color.propTypes,
  ...display.propTypes,
  ...height.propTypes,
  ...maxWidth.propTypes,
  ...space.propTypes,
  ...textAlign.propTypes,
  ...width.propTypes
};

Box.section = Box.withComponent('section');

export default Box;
