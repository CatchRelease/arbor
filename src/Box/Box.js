import styled from '@emotion/styled';
import {
  borders,
  borderColor,
  borderRadius,
  boxShadow,
  color,
  display,
  position,
  space,
  width,
  height,
  maxHeight,
  maxWidth,
  minWidth,
  overflow,
  textAlign
} from 'styled-system';

const Box = styled.div`
  box-sizing: border-box;
  ${borders};
  ${borderColor};
  ${borderRadius};
  ${boxShadow};
  ${color};
  ${display};
  ${height};
  ${maxHeight};
  ${maxWidth};
  ${minWidth};
  ${overflow};
  ${position};
  ${space};
  ${textAlign};
  ${width};
`;

Box.propTypes = {
  ...borders.propTypes,
  ...borderColor.propTypes,
  ...boxShadow.propTypes,
  ...borderRadius.propTypes,
  ...color.propTypes,
  ...display.propTypes,
  ...height.propTypes,
  ...maxHeight.propTypes,
  ...maxWidth.propTypes,
  ...minWidth.propTypes,
  ...overflow.propTypes,
  ...position.propTypes,
  ...space.propTypes,
  ...textAlign.propTypes,
  ...width.propTypes
};

export default Box;
