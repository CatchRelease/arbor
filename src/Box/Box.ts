import { ComponentProps } from 'react';
import styled from '@emotion/styled';
import {
  borders,
  BordersProps,
  boxShadow,
  BoxShadowProps,
  display,
  DisplayProps,
  gridArea,
  GridAreaProps,
  height,
  HeightProps,
  maxHeight,
  minHeight,
  position,
  PositionProps,
  width,
  WidthProps
} from 'styled-system';

import { PolyComponent } from '../polyComponent';

import Text, { TextProps } from '../Text';
import { WithColorPropFix } from '../colorPropFix';

type InternalProps = TextProps &
  BordersProps &
  BoxShadowProps &
  DisplayProps &
  HeightProps &
  GridAreaProps &
  PositionProps &
  WidthProps;

const TextDiv = Text.withComponent('div');
const TextDivWithColorPropFix: WithColorPropFix<typeof TextDiv> = TextDiv;

const Box = styled(TextDivWithColorPropFix)<InternalProps>`
  box-sizing: border-box;
  ${borders};
  ${boxShadow};
  ${display};
  ${height};
  ${gridArea};
  ${maxHeight};
  ${minHeight};
  ${position};
  ${width};
`;

const PolyBox: PolyComponent<'div', InternalProps> = Box;

export type Props = ComponentProps<typeof PolyBox>;

export default Object.assign(PolyBox, {
  withComponent: Box.withComponent
});
