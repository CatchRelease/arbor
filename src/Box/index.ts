export { default, Props as BoxProps } from './Box';
