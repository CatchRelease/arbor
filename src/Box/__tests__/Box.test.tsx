import { shallow } from 'enzyme';

import Box, { Props as BoxProps } from '../Box';

describe('Box', () => {
  it('supports appropriate element props based on "as" prop', () => {
    expect(shallow(<Box as="a" href="lol" />)).toMatchSnapshot();
    expect(shallow(<Box as="img" src="lol" />)).toMatchSnapshot();
  });

  it('supports generic style props such as borderTop with or without "as"', () => {
    expect(shallow(<Box borderTop="42" />)).toMatchSnapshot();
    expect(shallow(<Box as="a" borderTop="42" />)).toMatchSnapshot();
  });

  it('exports props compatible with itself', () => {
    const props: BoxProps = { p: 'regular' };

    const tree = shallow(<Box {...props}>Lalala</Box>);

    expect(tree).toMatchSnapshot();
  });
});
