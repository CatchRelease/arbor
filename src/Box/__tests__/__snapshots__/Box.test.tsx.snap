// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box exports props compatible with itself 1`] = `
.emotion-0 {
  padding: regular;
  box-sizing: border-box;
}

<Fragment>
  <Noop />
  <div
    className="emotion-0"
  >
    Lalala
  </div>
</Fragment>
`;

exports[`Box supports appropriate element props based on "as" prop 1`] = `
.emotion-0 {
  box-sizing: border-box;
}

<Fragment>
  <Noop />
  <a
    className="emotion-0"
    href="lol"
  />
</Fragment>
`;

exports[`Box supports appropriate element props based on "as" prop 2`] = `
.emotion-0 {
  box-sizing: border-box;
}

<Fragment>
  <Noop />
  <img
    className="emotion-0"
    src="lol"
  />
</Fragment>
`;

exports[`Box supports generic style props such as borderTop with or without "as" 1`] = `
.emotion-0 {
  box-sizing: border-box;
  border-top: 42;
}

<Fragment>
  <Noop />
  <div
    className="emotion-0"
  />
</Fragment>
`;

exports[`Box supports generic style props such as borderTop with or without "as" 2`] = `
.emotion-0 {
  box-sizing: border-box;
  border-top: 42;
}

<Fragment>
  <Noop />
  <a
    className="emotion-0"
  />
</Fragment>
`;
